app/controllers/users_controller.rb

```diff
@@ -22,11 +22,19 @@ def edit
     redirect_to login_path unless current_user 
   end
 
-  def patch
-    @user = User.update(user_params)
-    if @user.save
-      redirect_to profile_path(@user)
+  def update
+    # Only accept password change if both fields are filled in
+    if  params[:user][:password].blank? || 
+        params[:user][:password_confirmation].blank?
+    # Otherwise, remove both from params
+      params[:user].delete(:password)
+      params[:user].delete(:password_confirmation)
+    end
+    if current_user.update_attributes(user_params)
+      flash[:success] = "Your profile has been updated."
+      redirect_to profile_path
     else
```

Comment:
@iandouglas @dionew1 

I can get in here with a pry in development by loading the page and clicking Update User without changing any of the fields.  The test should do the same (visit profile_edit_path, click_on 'Update User'), but it redirects to profile_path and flashes a success message.

Reply:
Are you mocking the login in your test? If so, the current_user object won't return any chances you make to the user.